test/test_torch.py

```diff
@@ -6082,8 +6082,13 @@ def compare_strides(s1, s2, div):
                 _test_helper(x, op, unary=True)
 
     @skipMeta
-    def test_dlpack_conversion(self, device):
-        x = torch.randn(1, 2, 3, 4, device=device, dtype=torch.float)
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_dlpack_conversion(self, device, dtype):
```

Comment:
If we wanted to expand dlpack testing, how would we do it? Not to say that needs to be part of this PR, but I'm curious how we could do it. Here we're testing that the "from_dlpack" operation is the inverse operation of "to_dlpack," but a valid implementation of both operations (w.r.t. this test) would be `lamda x : x`. Is there any additional validation we can run on `to_dlpack` tensor to ensure it's 'dlpacked' properly? Similarly, is there a reference implementation we can create a 'dlpacked' tensor with to ensure that `from_dlpack` translates it to a torch tensor correctly?

Reply:
A way to work around this is to add a consumer by either using a third-party library such as CuPy, or creating some utility code that checks for the binary values stored in packed tensors and whether they are conformant or not to the actual tensor that was consumed/produced, but I think this is code that needs to live in ATen and be exposed to python through the `testing` package.

Reply:
Maybe an interesting task for the future.